.github/workflows/ios.yml

```diff
@@ -0,0 +1,181 @@
+
+    steps:
+      - name: Set xcode version
+        run: sudo xcode-select -s /Applications/Xcode.app
+
+      - name: Set ref
+        run: |
+          if [[ "${{ github.event.inputs.ref_name }}" != "" ]]; then
+            echo "REF_NAME=${{ github.event.inputs.ref_name }}" >> $GITHUB_ENV
+          else
+            echo "REF_NAME=${{ github.ref_name }}" >> $GITHUB_ENV
+          fi
+
+      - uses: actions/checkout@v5
+        with:
+          ref: ${{ env.REF_NAME }}
+
+      - name: Set env
+        run: |
+          echo "UPLOAD_TO_TESTFLIGHT=${{ github.event.inputs.upload_to_testflight }}" >> $GITHUB_ENV
+          export IOS_SCHEME="${{ github.event.inputs.scheme }}"
+          ./scripts/gh-ios-env >> $GITHUB_ENV
+
+      - name: Decode and save sentry.properties
+        run: |
+          echo ${{ env.SENTRY_PROPERTIES_CONTENT }} | base64 --decode > ./ios/sentry.properties
+
+      # Most of the vars below are set on the ./scripts/gh-ios-env script.
+      - name: Debug info
+        run: |
+          echo "UPLOAD_TO_TESTFLIGHT=${{ github.event.inputs.upload_to_testflight }}"
+          echo "BUILD_VERSION=${BUILD_VERSION}"
+          echo "REF_NAME=${REF_NAME}"
+          echo "APP_VERSION=${APP_VERSION}"
+          echo "APP_ID=${APP_ID}"
+          echo "APP_NAME=${APP_NAME}"
+          echo "IOS_SCHEME=${IOS_SCHEME}"
+          echo "FASTLANE_LANE=${FASTLANE_LANE}"
+
+          echo "SENTRY_PROPERTIES_CONTENT=$(echo -n $SENTRY_PROPERTIES_CONTENT | md5)" 
+          echo "APPLE_CONNECT_KEY_CONTENT=$(echo -n $APPLE_CONNECT_KEY_CONTENT | md5)"
+          echo "FASTLANE_GIT_DEPLOY_KEY=$(echo -n $FASTLANE_GIT_DEPLOY_KEY | md5)"
+
+      - name: Make scripts executable
+        run: find scripts -type f -exec chmod +x {} \;
+
+      - name: Set up Node
+        uses: actions/setup-node@v6
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+          cache: "yarn"
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ env.RUBY_VERSION }}
+          bundler-cache: true
+
+      - name: Cache Cocoapods
+        uses: actions/cache@v4
+        id: cocoapods-cache
+        with:
+          path: ios/Pods
+          key: ${{ runner.os }}-pods-${{ hashFiles('ios/Podfile.lock') }}
+
+      - name: Run Yarn Install
+        run: |
+          yarn install --frozen-lockfile --immutable
+
+      - name: Run Post Install
+        run: |
+          yarn postinstall
+
+      - name: Build and Distribute Application
+        id: fastlane
+        run: |
+          bundle exec fastlane --version
+          bundle exec fastlane ios ${{ env.FASTLANE_LANE }}
+
+      # Dev builds: Short retention for rapid iteration
+      - name: Upload dev build artifacts
+        uses: actions/upload-artifact@v4
+        if: always() && env.IOS_SCHEME == 'freighter-mobile-dev'
+        with:
+          name: ios-build-dev-${{ env.BUILD_VERSION }}
+          path: /tmp/build
+          retention-days: 7
```

Comment:
Dev build binaries are retained for 7 days only since we have quick (and not critical) iterations on it